src/algorithms/toolbox.jl

```diff
@@ -94,3 +94,37 @@ function MPSKit.correlation_length(peps::InfinitePEPS, env::CTMRGEnv; num_vals=2
 
     return ξ_h, ξ_v, λ_h, λ_v
 end
+
+"""
+    product_peps(peps_args...; unitcell=(1, 1), noise_amp=1e-2, state_vector=nothing)
+
+Initialize a normalized random product PEPS with noise. The given arguments are passed on to
+the `InfinitePEPS` constructor.
+
+The noise intensity can be tuned with `noise_amp`. The product state coefficients can be
+specified using the `state_vector` kwarg in the form of a matrix of size `unitcell`
+containing vectors that match the PEPS physical dimensions. If `nothing` is provided,
+random Gaussian coefficients are used.
+"""
+function product_peps(peps_args...; unitcell=(1, 1), noise_amp=1e-2, state_vector=nothing)
+    noise_peps = InfinitePEPS(peps_args...; unitcell)
+    typeof(spacetype(noise_peps[1])) <: GradedSpace &&
+        error("symmetric tensors not generically supported")
+    if isnothing(state_vector)
+        state_vector = map(noise_peps.A) do t
+            randn(scalartype(t), dim(space(t, 1)))
+        end
+    else
+        all(dim.(space.(noise_peps.A, 1)) .== length.(state_vector)) ||
+            throw(ArgumentError("state vectors must match the physical dimension"))
+    end
+    prod_tensors = map(zip(noise_peps.A, state_vector)) do (t, v)
+        pt = zero(t)
+        ptdata = block(pt, Trivial())
+        ptdata[:, 1, 1, 1, 1] .= v
```

Comment:
Just as a small aside, this will work, but I think you might be misinterpreting this:
`block` will return a Matrix, which is the reshaped view of the tensor, the matrix representation of the linear map. Then, this is using the fact that julia is capable of indexing with trailing ones:
```julia
julia> A = zeros(3)
3-element Vector{Float64}:
 0.0
 0.0
 0.0

julia> A[1,1,1,1,1]
0.0
```
so this is completely equivalent to `ptdata[:, 1] .= v`, because of Julia magic.

If you want to access the "tensor data", i.e. as a N-dimensional array, for trivial tensors this can be achieved as `pt[]`, which gives a 5d array (view) into the tensor data. `pt[][:, 1, 1, 1, 1] .= v` would then be more or less what I think you wanted to do.

Again, because of the specifics of this case, here it boils down to the same thing.

Reply:
Thanks for the tip, I wasn't aware. Then I would prefer using `pt[]` to make the indexing a bit more explicit in this case.